providers/base/bin/reboot_check_test.sh

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+
+device_check="true"
+service_check="true"
+
+service_check() {
+    echo -e "\n## Check system service.."
+    COUNT=$(systemctl --system --no-ask-password --no-pager --no-legend list-units --state=failed | wc -l)
+    printf "Found %s failed units\n" "$COUNT"
+    if [ "$COUNT" != 0 ]; then
+        printf "\nFailed units:\n"
+        systemctl --system --no-ask-password --no-pager list-units --state=failed
+        service_check="false"
+    fi
+}
+
+dump() {
+    echo "## Dump the devices information to $output_dir.."
+    mkdir -p "$output_dir"
+    lspci -i "$SNAP"/usr/share/misc/pci.ids > "$output_dir"/lspci_log
+    iw dev | grep "Interface\|addr\|ssid" > "$output_dir"/wifi_conn
+    checkbox-support-lsusb -f "$CHECKBOX_RUNTIME"/var/lib/usbutils/usb.ids -s | sort > "$output_dir"/lsusb_log
+    echo "dump devices complete"
+    sync
+}
+
+compare() {
+    echo -e "\n## Compare the devices.."
+    if ! diff -u "$compare_dir"/lspci_log "$output_dir"/lspci_log; then
+        echo "lspci mismatch during cycle"
```

Comment:
Similar to previous comment, it would be nice to have a more information message, something like:

> The list of PCI devices (lspci) is different from the original list gathered at the beginning of the session!

Same goes for other error messages below.